1.20/src/main/java/org/infernalstudios/nebs/NekosEnchantedBooks.java

```diff
@@ -1,41 +1,56 @@
 package org.infernalstudios.nebs;
 
-import net.minecraft.data.DataGenerator;
-import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.data.event.GatherDataEvent;
-import net.minecraftforge.eventbus.api.IEventBus;
-import net.minecraftforge.fml.DistExecutor;
 import net.minecraftforge.fml.IExtensionPoint;
 import net.minecraftforge.fml.ModLoadingContext;
 import net.minecraftforge.fml.common.Mod;
 import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
-import net.minecraftforge.fml.loading.FMLEnvironment;
 import net.minecraftforge.network.NetworkConstants;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+/**
+ * <h1>Neko's Enchanted Books</h1>
+ *
+ * This is the main class for the Neko's Enchanted Books (shortened to NEBs) mod, loaded by Forge. The mod itself does
+ * not interface much with Forge itself, but rather uses
+ * {@link org.infernalstudios.nebs.mixin.BlockModelMixin BlockModelMixin} to inject the custom item overrides for the
+ * enchanted books provided in {@link EnchantedBookOverrides}.
+ */
 @Mod(NekosEnchantedBooks.MOD_ID)
 public class NekosEnchantedBooks {
+    /** The Mod ID for this mod. Note that this variable is in-lined at compile time, so it is safe to reference. */
     public static final String MOD_ID = "nebs";
-    public static final Logger LOGGER = LogManager.getLogger();
+    /** The logger for this mod. Package-private since it doesn't need to be accessed in many places. */
+    static final Logger LOGGER = LogManager.getLogger();
 
+    /**
+     * The constructor for the mod. This does two things:
+     *
+     * <ol>
+     *     <li>Register the display test extension point, which tells the game to ignore this mod when polling servers
+     *     for mod compatibility.</li>
+     *     <li>Add our data generator as a listener to the {@link GatherDataEvent}. See
+     *     {@link #gatherData(GatherDataEvent)}</li>
+     * </ol>
+     */
     public NekosEnchantedBooks() {
+        // Newer Forge versions make this simpler, but here the old way is used for backwards compatibility with older Forge versions
         ModLoadingContext.get().registerExtensionPoint(IExtensionPoint.DisplayTest.class,
                 () -> new IExtensionPoint.DisplayTest(() -> NetworkConstants.IGNORESERVERONLY, (a, b) -> true));
 
-        // Only load the mod on the client side
-        if (FMLEnvironment.dist == Dist.CLIENT) {
-            IEventBus bus = FMLJavaModLoadingContext.get().getModEventBus();
-            bus.addListener(this::gatherData);
-        }
+        FMLJavaModLoadingContext.get().getModEventBus().addListener(this::gatherData);
```

Comment:
We could change these to take in the mod loading context from the constructor, as both `ModLoadingContext.get()` and `FMLJavaModLoadingContext.get()` are deprecated IIRC. Correct me if I'm wrong.

Reply:
That is in a future Forge version, which this mod does not use. Do you want me to bump the Forge version? It would require me to bump the required version in mods.toml.

Reply:
Oh, sorry, didn't know that